### DISCUSSION TITLE
Controlling maximum active devices

### DISCUSSION THREAD

**Discussion by jugrajsingh** [zitadel/zitadel#3877] [Ideas]
Is there a way to control how many devices are active under one user.

The aim is to control number of devices a user can parallellely connected to at organisation level

Here is a scenario
Consider organisation has maximum device limit of 3 if we user is already locked in two three different browser on the same machine or different machines. Any additional machine the user ads should prompt user to log out of any of the other three active sessions

**Comment by fforootd** [comment]
Hi @jugrajsingh 

I like the idea you are having there. This could be implemented by creating a new policy which applies such a setting to the whole system and to organizations.

One thought that crosses my mind though is that maybe there should be some kind of distinction based on different parameters (type of user-agent, client, authenticator type, ...)

**Reply by fforootd** [reply]
Btw. while thinking about this... such a limit could also be interesting to apply on clients (oauth, saml, whatever)

**Reply by jugrajsingh** [reply]
> Hi @jugrajsingh
> 
> I like the idea you are having there. This could be implemented by creating a new policy which applies such a setting to the whole system and to organizations.

Is there a documentation that I can refer to test this suggestion out

> 
> One thought that crosses my mind though is that maybe there should be some kind of distinction based on different parameters (type of user-agent, client, authenticator type, ...)

I think there is a concept of browser finger printing that will help us distinguish between if I user is login between multiple profiles of browsers 
so we don't have to identity file user agent and other parameters just the simple fingerprint generated by the library

**Comment by jugrajsingh** [comment]
Hello, 

I am interested in implementing controls for each login attempt. Specifically, I would like to achieve device limiting behavior similar to that used by Netflix. Could someone please offer guidance on how to accomplish this? Thank you.

**Reply by hifabienne** [reply]
Hei @jugrajsingh 
We do not provide something specific for device limitation. But you can use a combination of user metadata and actions.
User metadata are basically key values on users. You can store the devices in the user metadata.
The actions are javascript snippets you can implement and run at specific points. So you could implement an action that checks if there is a new device, adds it to the metadata or sends an error.
